### PR TITLE
feat: update API requests with OpenAPI generated client

### LINKED ISSUE #3386: dashboard(apiserver): swag generates random definitions in different builds
## Bug Report

**What version of Kubernetes are you using?**

**What version of Chaos Mesh are you using?**

**What did you do?**

Run `./hack/generate_swagger_spec.sh` in the root folder.

**What did you expect to see?**

Definitions keep the same in different builds.

**What did you see instead?**

image

The `github.com_chaos-mesh_chaos-mesh_` prefix will be added and removed in a continuous build.

**Output of chaosctl**

### PR DESCRIPTION
**Pull request by g1eny0ung**
### What problem does this PR solve?

Close #3386

### What's changed and how it works?

Proposal: [rfc: OpenAPI to TypeScript API Client and Forms](https://github.com/chaos-mesh/rfcs/pull/30)

This PR updates nearly all API requests (Some with strong dependency bindings are ignored and will solve next time) with the OpenAPI generated client. **Note that this PR must be merged after #2770 because it's based on it.**

The overall update process is as follows:

1. Upgrade `swag` to the latest version to meet some needs (fast generation).
2. Run `./hack/generate_swagger_spec.sh` in your local to generate swagger files.
3. Then run `yarn workspace @ui/openapi codegen client` to update the client.
4. Finally, the API requests are updated one by one according to the type inference of TypeScript.

### Related changes

- [ ] Need to update `chaos-mesh/website`
- [ ] Need to update `Dashboard UI`
- Need to **cheery-pick to release branches**
  - [ ] release-2.1
  - [ ] release-2.0

### Checklist

Tests

- [ ] Unit test
- [ ] E2E test
- [x] No code
- [ ] Manual test (add steps below)

Side effects

- [ ] Breaking backward compatibility

### DCO

If you find the DCO check fails, please run commands like below (Depends on the actual situations. For example, if the failed commit isn't the most recent) to fix it:

```shell
git commit --amend --signoff
git push --force
```

### CODE REVIEW

File: ui/packages/openapi/utils.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+/**
+ * Visit all children in given node and filter them.
+ *
+ * @export
+ * @param {ts.Node} node
+ * @param {ts.Node[]} nodes
+ * @param {filterCallbackFn} filterCallbackFn
+ */
+export function visitFilter(node, nodes, filterCallbackFn) {
```
[the remark this answers is not in the document]
Comment (g1eny0ung) [reply ui/packages/openapi/utils.js:32]: ignore

[the remark this answers is not in the document]
Comment (g1eny0ung) [reply ui/packages/openapi/utils.js:32]: ignore

**Comment by g1eny0ung** [comment]
Refer to https://github.com/swaggo/swag/pull/1108, maybe soon we can rely on `swag`'s next release to merge this PR. So I would like to hold this PR for a while.

**Comment by STRRL** [comment]
Hi @g1eny0ung , I tried to resolve the conflict by regenerating the swagger.yaml.

But it seems `swag` failed to parse import alias with `u "github.com/chaos-mesh/chaos-mesh/pkg/dashboard/apiserver/utils"`:

```
2022/04/25 07:09:07 ParseComment error in file pkg/dashboard/apiserver/template/template.go :can not find schema type: "u.APIError"
```

**Comment by g1eny0ung** [comment]
Note: To solve #3386, I move all inline types in every sub-package under `apiserver` into the sub package `types`. This can prevent `swag` generate the type name automatically.

**Comment by g1eny0ung** [comment]
> Hi @g1eny0ung , I tried to resolve the conflict by regenerating the swagger.yaml.
> 
> But it seems `swag` failed to parse import alias with `u "github.com/chaos-mesh/chaos-mesh/pkg/dashboard/apiserver/utils"`:
> 
> ```
> 2022/04/25 07:09:07 ParseComment error in file pkg/dashboard/apiserver/template/template.go :can not find schema type: "u.APIError"
> ```

@STRRL 😵 Sorry I missed this comment. It works now.

**Pull request merged**